MVVM/ViewModel/ZedMonitor.cs

```diff
@@ -55,29 +74,50 @@ public Task OnGyroscopeChanged(string subTopic, MqttApplicationMessage? msg)
 			return Task.CompletedTask;
         }  
     }
+
+    public void _on_timer_timeout()
+    {
+        timeSLU += timerInterval;
+
+        if (timeSLU == errorTime)
+        {
+            MainViewModel.EventLogger?.LogMessage($"ZedMonitor Error: gyro data >{errorTime} seconds old.");
+
+        }
+        if (timeSLU >= errorTime)
+        {
+            errorDisplay.Visible = true;
+        }
+        TimeUpdate(timeSLU);
+    }
+
     public Quaternion PullGyroscope(MqttApplicationMessage? msg)
     {
         try
         {
             Gyroscope = JsonSerializer.Deserialize<MqttClasses.ZedImuData>(msg.ConvertPayloadToString());
             Quaternion Quat = new Quaternion((float)Gyroscope.orientation.x, (float)Gyroscope.orientation.y, (float)Gyroscope.orientation.z, (float)Gyroscope.orientation.w);
+            error = false;
+
             return Quat;
         }
         catch (Exception e)
         {
-            GD.Print($"ZedMonitor Error (Something is wrong with json/deserialization): {e.Message}");
-            return new Quaternion();
+            MainViewModel.EventLogger?.LogMessage($"ZedMonitor Error (Something is wrong with json/deserialization): {e.Message}");
+            error = true;
+            return new Quaternion((float)Gyroscope.orientation.x, (float)Gyroscope.orientation.y, (float)Gyroscope.orientation.z, (float)Gyroscope.orientation.w);

```

Comment:
I think `Gyroscope` will be null when json deserialization fails.
```suggestion
            return new Quaternion.Identity;
```

Reply:
I checked it with a hivemq web client and gyroscope retains previous, correctly deserialized floats. However I agree that returning the same thing again is pointless especially now when we do nothing with that Quaternion. Changed in [c5fe7fa](https://github.com/wisniax/RoverControlApp/pull/10/commits/c5fe7fac5e284ba701c9156a7e23a590a999a66d)